Framework/Algorithms/inc/MantidAlgorithms/MergeRuns.h

```diff
@@ -128,18 +128,20 @@ class DLLExport MergeRuns : public API::MultiPeriodGroupAlgorithm {
   using Mantid::API::Algorithm::validateInputs;
   bool validateInputsForEventWorkspaces(
       const std::vector<std::string> &inputWorkspaces);
-  void calculateRebinParams(const API::MatrixWorkspace_const_sptr &ws1,
-                            const API::MatrixWorkspace_const_sptr &ws2,
-                            std::vector<double> &params) const;
-  void noOverlapParams(const HistogramData::HistogramX &X1,
-                       const HistogramData::HistogramX &X2,
-                       std::vector<double> &params) const;
-  void intersectionParams(const HistogramData::HistogramX &X1, int64_t &i,
-                          const HistogramData::HistogramX &X2,
-                          std::vector<double> &params) const;
-  void inclusionParams(const HistogramData::HistogramX &X1, int64_t &i,
-                       const HistogramData::HistogramX &X2,
-                       std::vector<double> &params) const;
+  std::pair<bool, std::vector<double>> checkRebinning();
```

Comment:
Probably just a matter of taste, but calls to `checkRebinning` might look a bit nicer if you returned a `boost::optional<std::vector<double>>` here, which would avoid the annoying `std::tie` at the other end - you could rename the method something like `getRebinParamsIfNeeded` to make it clear what's going on (plus the return value could be `const`)

Reply:
Forgot completely about `boost::optional`! I was trying to use `std::optional` here but it's C++17.